Source/AwsCommonRuntimeKit/auth/credentials/CRTAWSCredentialsProvider.swift

```diff
@@ -18,11 +18,13 @@ public final class CRTAWSCredentialsProvider {
     }
 
     public convenience init(fromProvider impl: CRTCredentialsProvider,
-                            shutDownOptions: CRTCredentialsProviderShutdownOptions? = nil,
-                            allocator: Allocator = defaultAllocator) throws {
-        let shutDownOptions = CRTAWSCredentialsProvider.setUpShutDownOptions(shutDownOptions: shutDownOptions)
+                            allocator: Allocator = defaultAllocator,
+                            shutdownCallback: ShutdownCallback? = nil) throws {
+        let shutdownCallbackCore = ShutdownCallbackCore(shutdownCallback)?
+                .getRetainedCredentialProviderShutdownOptions()
```

Comment:
if `aws_credentials_provider_new_delegate()` fails, we never `releaseBecauseThisCallbackWillNeverFire()`

Reply:
Thanks, fixed.